go.work

```diff
@@ -1,4 +1,4 @@
-go 1.23.5
+go 1.23.6
```

Comment:
I am not even sure we need this go.work following the removal of all the modules in favour of packages.
Approving here, will evaluate dropping this file in a separate PR